src/core/masking.ts

```diff
@@ -194,7 +201,17 @@ function maskSelectionSet(
           ];
         }
         case Kind.FRAGMENT_SPREAD: {
-          const fragment = context.fragmentMap[selection.name.value];
+          const fragmentName = selection.name.value;
+          const fragment: FragmentDefinitionNode | null =
+            context.fragmentMap[fragmentName] ||
+            context.cache.lookupFragment(fragmentName);
+
```

Comment:
```suggestion
          let fragment: FragmentDefinitionNode | null =
            context.fragmentMap[fragmentName] ||
            (context.fragmentMap[fragmentName] =
              context.cache.lookupFragment(fragmentName)!);
```

Let's cache this `.lookupFragment` call, WDYT?

Reply:
Good call! I dig it.